src/MediaWiki.ts

```diff
@@ -54,24 +56,37 @@ class MediaWiki {
 
   public async login(downloader: Downloader) {
     if (this.username && this.password) {
-      let url = `${this.apiUrl.href}action=login&format=json&lgname=${this.username}&lgpassword=${this.password}`;
+      let url = this.apiUrl.href;
       if (this.domain) {
         url = `${url}&lgdomain=${this.domain}`;
       }
-      const { content } = await downloader.downloadContent(url);
-      let body = content.toString();
-      let jsonResponse = JSON.parse(body).login;
-      downloader.loginCookie = `${jsonResponse.cookieprefix}_session=${jsonResponse.sessionid}`;
-      if (jsonResponse.result !== 'SUCCESS') {
-        url = `${url}&lgtoken=${jsonResponse.token}`;
-        const { content: subContent } = await downloader.downloadContent(url);
-        body = subContent.toString();
-        jsonResponse = JSON.parse(body).login;
-        if (jsonResponse.result !== 'Success') {
+      // Getting token to login.
+      const {content, responseHeaders}  = await downloader.downloadContent(url + 'action=query&meta=tokens&type=login&format=json');
+
+      // Logging in
+      await axios(this.apiUrl.href, {
+        data: qs.stringify({
+          action: 'login',
+		      format: 'json',
+          lgname: this.username,
+          lgpassword: this.password,
+          lgtoken: JSON.parse(content.toString()).query.tokens.logintoken,
+        }),
+        headers: {
+          Cookie: responseHeaders['set-cookie'].join(';'),
+          'Content-Type': 'application/x-www-form-urlencoded'
+        },
+        method: 'POST',
+      })
+      .then((resp) => {
+        if (resp.data.login.result !== 'Success') {
           throw new Error('Login Failed');
         }
-        downloader.loginCookie = `${jsonResponse.cookieprefix}_session=${jsonResponse.sessionid}`;
-      }
+        downloader.arrayBufferRequestOptions.headers.Cookie = resp.headers['set-cookie'].join(';');
```

Comment:
@kelson42 I set cookies here so that all request to api are being made properly was I supposed to do anything else?

Reply:
@MananJethwani Which piece o code secures this cookie is use for all requests?

Reply:
` downloader.arrayBufferRequestOptions.headers.Cookie = resp.headers['set-cookie'].join(';');`
I am not sure that this works yet still checking on private wiki but mostprobably should

Reply:
sorry my bad should have changed `downloader.loginCookie`